numpy/_core/tests/test_datetime.py

```diff
@@ -585,6 +585,7 @@ def test_datetime_nat_casting(self):
         cnan = nan.astype('D')
```

Comment:
One thing that might be cool, would be to split this into two tests and use `pytest.parametrize`.  But it honestly doesn't really need to be part of this PR.

Reply:
Thank you for the review! I will add additional PRs to improve and expand the tests, and consider one to port the datetime casts.